core/src/test/java/org/elasticsearch/search/scriptfilter/ScriptQuerySearchIT.java

```diff
@@ -87,6 +103,57 @@ public Settings indexSettings() {
                 .build();
     }
 
+    public void testCustomScriptBinaryField() throws Exception {
+        final byte[] randomBytesDoc1 = getRandomBytes(15);
+        final byte[] randomBytesDoc2 = getRandomBytes(16);
+
+        assertAcked(
+                client().admin().indices().prepareCreate("my-index")
+                        .addMapping("my-type", createMappingSource("binary"))
+                        .setSettings(indexSettings())
+        );
+        client().prepareIndex("my-index", "my-type", "1")
+                .setSource(jsonBuilder().startObject().field("binaryData",
+                        Base64.getEncoder().encodeToString(randomBytesDoc1)).endObject())
+                .get();
+        flush();
+        client().prepareIndex("my-index", "my-type", "2")
+                .setSource(jsonBuilder().startObject().field("binaryData",
+                        Base64.getEncoder().encodeToString(randomBytesDoc2)).endObject())
+                .get();
+        flush();
+        refresh();
+
+        SearchResponse response = client().prepareSearch()
+                .setQuery(scriptQuery(
+                        new Script(ScriptType.INLINE, CustomScriptPlugin.NAME, "doc['binaryData'].get(0).length > 15", Collections.emptyMap())))
+                .addScriptField("sbinaryData",
+                        new Script(ScriptType.INLINE, CustomScriptPlugin.NAME, "doc['binaryData'].get(0).length", Collections.emptyMap()))
+                .get();
+
+        assertThat(response.getHits().totalHits(), equalTo(1L));
+        assertThat(response.getHits().getAt(0).id(), equalTo("2"));
+        assertThat(response.getHits().getAt(0).fields().get("sbinaryData").values().get(0), equalTo(16));
+
+    }
+
+    private byte[] getRandomBytes(int len) {
+        final byte[] randomBytes = new byte[len];
+        new Random().nextBytes(randomBytes);
```

Comment:
This API call is forbidden and fails the build. `random().nextBytes(randomBytes);` is fine though. I'll fix it before I merge.